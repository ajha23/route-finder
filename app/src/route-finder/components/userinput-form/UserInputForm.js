import React, { Component } from 'react';
import { maps } from '../../../common/services';

import './UserInputForm.scss';

/**
 * Component Name: UserInputForm
 * Description: This component will render the auto complete form
 */
class UserInputForm extends Component {
    originInput;
    originInputAutoComplete;
    destInput;
    destInputAutoComplete;

   
    getRoute = () => {
        const origin = this.originInputAutoComplete.getPlace();
        const dest = this.destInputAutoComplete.getPlace();
        this.props.getDirections(origin, dest);
    };

  
    renderAutoComplete = async () => {
        const maps = await this.props.maps();

        this.originInputAutoComplete = new maps.places.Autocomplete(this.originInput);

        this.destInputAutoComplete = new maps.places.Autocomplete(this.destInput);
    };

   
    componentDidMount() {
        this.renderAutoComplete();
    }

  
    render() {
        return (
            <div className="userinput-form">
                <div className="form-input">
                    <label>Starting Location</label>
                    <input type="text" ref={el => (this.originInput = el)} />
                </div>
                <div className="form-input">
                    <label>Drop-off point</label>
                    <input type="text" ref={el => (this.destInput = el)} />
                </div>
                <div className="go-btn">
                    <button onClick={this.getRoute}>Go</button>
                </div>
            </div>
        );
    }
}

UserInputForm.defaultProps = {
    maps
};

export default UserInputForm;
