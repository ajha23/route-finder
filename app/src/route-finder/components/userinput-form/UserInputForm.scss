@import '../../../responsive.scss';

.userinput-form {
    padding: 16px;
    display: flex;
    align-items: flex-end;

    @include mobile {
        flex-direction: column;
        align-items: flex-start;
    }
}

.form-input {
    margin-top: 16px;
    margin-right: 16px;

    > label {
        margin-bottom: 12px;
        font-size: 12px;
        display: block;
    }

    > input {
        display: block;
        padding: 8px;
        width: 200px;
    }
}

.go-btn > button {
    margin-top: 16px;
    padding: 10px;
    min-width: 100px;
    border: none;
    background: grey;
    color: #ffffff;
    cursor: pointer;
}
