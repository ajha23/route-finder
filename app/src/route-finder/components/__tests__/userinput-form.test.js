import React from 'react';
import renderer from 'react-test-renderer';

import UserInputForm from '../userinput-form/UserInputForm';

describe('Userinput Form Test', () => {
    it('render correctly', () => {
        const tree = renderer.create(<UserInputForm />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
