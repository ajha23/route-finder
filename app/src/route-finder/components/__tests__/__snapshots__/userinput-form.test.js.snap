// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Direction Form Test render correctly 1`] = `
<div
  className="userinput-form"
>
  <div
    className="form-input"
  >
    <label>
      Starting Location
    </label>
    <input
      type="text"
    />
  </div>
  <div
    className="form-input"
  >
    <label>
      Drop-off point
    </label>
    <input
      type="text"
    />
  </div>
  <div
    className="go-btn"
  >
    <button
      onClick={[Function]}
    >
      G0
    </button>
  </div>
</div>
`;

exports[`Userinput Form Test render correctly 1`] = `
<div
  className="userinput-form"
>
  <div
    className="form-input"
  >
    <label>
      Starting Location
    </label>
    <input
      type="text"
    />
  </div>
  <div
    className="form-input"
  >
    <label>
      Drop-off point
    </label>
    <input
      type="text"
    />
  </div>
  <div
    className="go-btn"
  >
    <button
      onClick={[Function]}
    >
      Go
    </button>
  </div>
</div>
`;
